packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-token.ts

```diff
@@ -44,3 +49,30 @@ defineProperty(Element.prototype, KEY__SHADOW_TOKEN, {
     },
     configurable: true,
 });
+
+defineProperty(Element.prototype, KEY__SHADOW_STATIC, {
+    set(this: Element, v: boolean) {
+        // Marking an element as static will propagate the shadow resolver to the children.
+        const fn = (this as any)[KEY__SHADOW_RESOLVER];
+        if (v) {
+            const treeWalker = createTreeWalker.call(
+                getOwnerDocument(this),
+                this,
+                NodeFilter.SHOW_ELEMENT | NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_TEXT,
```

Comment:
do we really need to link comments and texts? I thought it was only marking elements.

Reply:
yes, given that some templates may have comments.